faulty_code.py

```
@@ -0,0 +1,86 @@
+    """Calculate average. Bug: crashes on empty list, no type checking."""
+    total = 0
+    for n in numbers:
+        total += n
+    return total / len(numbers)
+
+
+def get_user_info(users, user_id):
+    """Get user by ID. Bug: KeyError if user_id missing, SQL injection risk."""
+    import sqlite3
+    conn = sqlite3.connect("users.db")
+    query = "SELECT * FROM users WHERE id = " + str(user_id)  # SQL injection
```

Ruff (0.15.13)

[error] Possible SQL injection vector through string-based query construction

(S608)